File: src/main/java/com/ennoblesoft/demo/controller/GlobalExceptionHandler.java
package com.ennoblesoft.demo.controller;

import com.ennoblesoft.demo.exception.CustomException;
import com.ennoblesoft.demo.response.ErrorResponse;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.ControllerAdvice;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.bind.annotation.ResponseBody;
import org.springframework.web.bind.annotation.ResponseStatus;

@ControllerAdvice
public class GlobalExceptionHandler {
    @ExceptionHandler(CustomException.class)
    @ResponseStatus(value = HttpStatus.NOT_FOUND)
    @ResponseBody
    public ErrorResponse showCustomException(CustomException ex) {
        return new ErrorResponse(ex.getErrorMessage());
    }
}

File: src/main/kotlin/com/ennoblesoft/springdemo/controller/GlobalExceptionHandler.kt
package com.ennoblesoft.springdemo.controller

import com.ennoblesoft.springdemo.exception.CustomException
import com.ennoblesoft.springdemo.response.ErrorResponse
import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.ControllerAdvice
import org.springframework.web.bind.annotation.ExceptionHandler
import org.springframework.web.bind.annotation.ResponseBody
import org.springframework.web.bind.annotation.ResponseStatus

@ControllerAdvice
class GlobalExceptionHandler {
    @ExceptionHandler(CustomException::class)
    @ResponseStatus(value = HttpStatus.NOT_FOUND)
    @ResponseBody
    fun showCustomException(ex: CustomException): ErrorResponse {
        return ErrorResponse(ex.errorMessage)
    }
}
